api/src/main/java/io/opentelemetry/common/ReadableAttributes.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.common;
+
+import javax.annotation.Nullable;
+
+/**
+ * A read-only container for String-keyed attributes.
+ *
+ * <p>See {@link Attributes} for the public API implementation.
+ */
+public interface ReadableAttributes {
+  /** The number of attributes contained in this. */
+  int size();
+
+  /** Whether there are any attributes contained in this. */
+  boolean isEmpty();
+
+  /** Iterates over all the key-value pairs of attributes contained by this instance. */
+  void forEach(KeyValueConsumer<AttributeValue> consumer);
+
+  /**
+   * Returns the value of the given key, or null if the key does not exist.
+   *
+   * <p>Warning: currently may be implemented via a linear search, depending on implementation, so
```

Comment:
Think we can remove this warning now. Seems too scary for what it actually is

Reply:
Be afraid, be very afraid.

Reply:
I'd be up for leaving the comment, but without the "Warning" word, to not scare people (yet have they understood the operation price ;) )

Reply:
Fixed.